### PR TITLE
Lint remaining specs in models

### PR DESCRIPTION
**Pull request by Renny4Real**
**WHY:**
Consistency & general housekeeping.
This is part two of PR #605 

**IN THIS PR:**
Lint and refactor the following files in the `spec/models folder`:

- `-spec/models/organisation_spec.rb`
- `spec/models/authorised_email_domain_spec.rb`
- `spec/models/user_spec.rb`

**Any suggestions/feedback would be appreciated.**

### CODE REVIEW

File: spec/models/organisation_spec.rb [hunk 1]
```diff
@@ -1,74 +1,79 @@
...
     end
 
-    context 'When an organisation signs up under the same name as another organisation' do
-      before { create(:organisation, name: 'Gov Org 1') }
+    it 'removes all associated ip addresses' do
+      expect { ip.reload }.to raise_error ActiveRecord::RecordNotFound
+    end
+  end
+
+  context 'when signing up an organisation with the same name as another registered organisation' do
```
Comment (tommotaylor) [inline spec/models/organisation_spec.rb]: Could easily be shortened to `'when name already exists'`

File: spec/models/organisation_spec.rb [hunk 2]
```diff
@@ -1,74 +1,79 @@
...
+  end
 
-    context 'When updating the organisations service email' do
-      it 'does not validate the organisations name' do
-        organisation = create(:organisation)
-        organisation.service_email = 'newserviceemail@gov.uk'
-        expect(organisation).to be_valid
-      end
+  context 'when updating an organisation service email' do
+    it 'remains valid' do
```
Comment (tommotaylor) [inline spec/models/organisation_spec.rb]: What does this actually test?

It creates an organisation, then updates the service email to a valid email and checks its valid? Seems unnecessary.

The original spec is `it 'does not validate the organisations name' do`. Which implies that an existing organisation with an invalid name, when updating the service email, does not validate the name again.

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
-  context 'validations' do
-    it { is_expected.to validate_presence_of(:name).on(:update) }
-  end
+  it { is_expected.to belong_to(:organisation).optional }
+  it { is_expected.to validate_presence_of(:name).on(:update) }
 
-  context 'permissions' do
-    context 'a new user' do
-      subject { create(:user) }
+  context 'when adding a new user' do
```
Comment (tommotaylor) [inline spec/models/user_spec.rb]: This context description doesn't really describe the context of the two specs that follow it. Something more descriptive would be `'when creating a user without explicit permissions'`

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -1,23 +1,16 @@
...
-        expect(subject).to be_can_manage_team
-      end
+    it 'receives default permission to manage team members' do
+      expect(user).to be_can_manage_team
+    end
 
-      it 'can manage locations' do
-        expect(subject).to be_can_manage_locations
-      end
+    it 'receives default permission to cmanage locations' do
```
Comment (tommotaylor) [inline spec/models/user_spec.rb]: spelling of "cmanage"

**Comment by Renny4Real** [comment]
UPDATE: Need the verify the necessity of the [test for updating the service email](https://github.com/alphagov/govwifi-admin/pull/608/files#diff-c7de48f70b756427e1be40127187ddbdR35) in organisation model spec file.

**Pull request merged**